src/use-cases/extract-tickets-use-case.ts

```diff
@@ -49,4 +55,15 @@ export class ConcreteExtractTicketsUseCase implements ExtractTicketsUseCase {
         map((ticketIdsCommits): ExtractTicketsOutput => ({ ticketIdsCommits }))
       );
   }
+
+  private extractCommits(input: ExtractTicketsInput): Observable<string[]> {
+    if (typeof input.reference === 'number') {
+      return this.pullRequestCommitExtractor.commits(
+        input.reference,
+        input.repository
+      );
+    } else {
+      return this.shaCommitExtractor.commits(input.reference, input.repository);
+    }
```

Comment:
That's where the choice is done for which commit extractor to use. 
Perhaps not the most pretty :thinking: But it does the job.